indico/modules/events/contributions/controllers/management.py

```diff
@@ -471,6 +472,25 @@ def _process(self):
         return send_file('book-of-abstracts.pdf', pdf.generate(), 'application/pdf')
 
 
+class RHContributionsImportCSV(RHManageContributionsActionsBase):
+    """Import contributions from a CSV file"""
+
+    def _process(self):
+        form = ImportContributionsForm()
+
+        if form.validate_on_submit():
+            contributions, changes = import_contributions_from_csv(self.event, form.source_file.data)
+            flash(ngettext("{} contribution has been imported.",
+                           "{} contributions have been imported.",
+                           len(contributions)).format(len(contributions)), 'success')
+            if changes:
+                flash(_("Event dates/times adjusted due to imported data."), 'warning')
+            return jsonify_data(flash=False, redirect=url_for('.manage_contributions', self.event),
+                                redirect_no_loading=True)
```

Comment:
I don't think `redirect_no_loading` is useful here. I added it for actions that trigger a download prompt where you usually remain on the same page and thus don't want to block it with the "Loading" dialog